static/js/composebox_typeahead.js

```diff
@@ -53,30 +53,57 @@ function query_matches_person(query, person) {
     // Case-insensitive.
     query = query.toLowerCase();
 
+    if (query.indexOf(" ") > 0) {
```

Comment:
Here is problem which conflicts with current behaviour under condition when a user has a name like `Extra User 0` and if we use `@User 0` it doesn't show typeahead(which does in current behaviour) because the `i` matches character parallelly in both `queries` and splitted `full_name` so index of User in `Extra User 0` is 1 whereas it is 0 in `User 0` and hence doesn't show typeahead.

Reply:
The solution here could be
```
        for (i = 0; i < queries.length - 1; i += 1) {
            if (person.full_name.toLowerCase().indexOf(queries[i]) === -1) {
                return false;
            }
        }
```
Does this solution makes sense

Reply:
We should only show the autocomplete for entries where first word in query matches the full first word of something being completed. So for `Extra user 0` if user enters `User 0` it won't autocomplete if `User 0` doesn't match with any typeahead. 
This solution(using `indexOf`) will not fulfill this criteria. i.e. if you enter `@ese i` then it will match with any word which include `ese`. 
The strategy I have used is when user press `" "` in query, then the query before `" "` should be exact match of some typeahead(using `===`) and query after `" "` can be part of typeahead(using `indexOf`).